thrust/thrust/detail/vector_base.inl

```diff
@@ -879,105 +879,155 @@ void vector_base<T, Alloc>::append(size_type n)
 template <typename T, typename Alloc>
 void vector_base<T, Alloc>::fill_insert(iterator position, size_type n, const T& x)
 {
-  if (n != 0)
+  // Early return if nothing to insert
+  if (n == 0)
   {
-    if (capacity() - size() >= n)
+    return;
```

Comment:
Sorry, I was not fully clear on the early return. We do need to clear the existing elements, so that the vector has a new size of 0